android/src/main/java/com/github/wuxudong/rncharts/charts/ChartBaseManager.java

```diff
@@ -396,19 +434,44 @@ protected void setCommonAxisConfig(Chart chart, AxisBase axis, ReadableMap propM
      */
     @ReactProp(name = "data")
     public void setData(Chart chart, ReadableMap propMap) {
-        chart.setData(getDataExtract(chart.getContext()).extract(propMap));
-        chart.invalidate();
+        ChartData data = getDataExtract(chart.getContext()).extract(propMap);
+        // data maybe null when the prop is removed temporarily from js side
```

Comment:
curious, when is data null? And if it's null temporarily, should we just not draw the chart in that instance? (e.g. added logic in my latest PR here to only draw the chart when `outdata` exists: https://github.com/indiemoney/snowstorm/blob/c26a85921adfd55dffb0bd81951f077f95c1b696/services/switchback/src/switchback/views/plan/viz.cljs#L368

Reply:
According to https://facebook.github.io/react-native/docs/native-components-android.html#3-expose-view-property-setters-using-reactprop-or-reactpropgroup-annotation
RN may pass in `null` (default value) when a prop is removed. I'm not sure how `:data` is removed from the initial props by us (maybe by accident? or some kind of interim state where we'll add it back eventually?) but I did see some times `null` got passed in here but there's *always* another call that sets the data correctly afterwards. Which make me think it could even be some kind of optimization that RN is doing behind the scene?